wasm2c/wasm-rt-impl.c

```diff
@@ -202,6 +235,41 @@ static void os_install_signal_handler(void) {
   }
 }
 
+static void os_disable_and_deallocate_altstack(void) {
+  /* in debug build, verify altstack allocated */
+#ifndef NDEBUG
+  if (!g_alt_stack) {
+    fprintf(stderr, "wasm-rt error: alternate stack is NULL\n");
+    abort();
+  }
+#endif
+
+  /* verify altstack was still in place */
+  stack_t ss;
+  if (sigaltstack(NULL, &ss) != 0) {
+    perror("sigaltstack failed");
+    abort();
+  }
+
+  if ((!g_alt_stack) || (ss.ss_flags & SS_DISABLE) ||
+      (ss.ss_sp != g_alt_stack) || (ss.ss_size != SIGSTKSZ)) {
+#ifndef NDEBUG
+    fprintf(stderr,
+            "wasm-rt warning: alternate stack was modified unexpectedly\n");
+#endif
```

Comment:
And this could just be `assert(false && ..`?

Reply:
I'd rather not make this one abort() on failure, since it seems plausible the application would install its own altstack at some point, and that shouldn't be a fatal error...